daml-foundations/daml-ghc/daml-compiler/src/DA/Service/Daml/Compiler/Impl/Handle.hs

```diff
@@ -153,15 +153,17 @@ newIdeState :: Options
             -> Logger.Handle IO
             -> Managed IdeState
 newIdeState compilerOpts mbEventHandler loggerH = do
-  mbScenarioService <- for mbEventHandler $ \eventHandler -> Scenario.startScenarioService eventHandler loggerH
-
-  -- Load the packages from the package database for the scenario service. We swallow errors here
-  -- but shake will report them when typechecking anything.
-  (_diags, pkgMap) <- liftIO $ CompilerService.generatePackageMap (optPackageDbs compilerOpts)
-  let rule = do
-        CompilerService.mainRule compilerOpts
-        Shake.addIdeGlobal $ GlobalPkgMap pkgMap
-  liftIO $ CompilerService.initialise rule mbEventHandler (toIdeLogger loggerH) compilerOpts mbScenarioService
+    mbScenarioService <-
+        for (guard (optScenarioService compilerOpts) >> mbEventHandler) $ \eventHandler ->
```

Comment:
It looks like the scenario service only uses the event handler to log exceptions and those are also written to the logger. So we can probably change this fairly easily to have `optScenarioService` fully control whether we start the scenario service instead of this somewhat weird behavior where it can only be used to disable but not enable the scenario service. Totally fine to do that in a separate PR.

Reply:
:+1: